trade-finance-manager-ui/server/api-ssrf.test.js

```diff
@@ -0,0 +1,537 @@
+    const mockResponse = 'Mock user';
+    beforeAll(() => {
+      mockAxios.reset();
+      const url = /^.*\/v1\/users\/.*$/;
+      mockAxios.onPut(url).reply(200, mockResponse);
+    });
+
+    it('Returns an error when a url traversal is supplied', async () => {
+      const urlTraversal = '../../../etc/stealpassword';
+      const expectedResponse = { status: 400, data: 'Invalid user id' };
+
+      const response = await api.updateUserPassword(urlTraversal, 'password update');
+
+      expect(response).toMatchObject(expectedResponse);
+    });
+
+    it('Returns an error when a local IP is supplied', async () => {
+      const localIp = '127.0.0.1';
+      const expectedResponse = { status: 400, data: 'Invalid user id' };
+
+      const response = await api.updateUserPassword(localIp, 'password update');
+
+      expect(response).toMatchObject(expectedResponse);
+    });
+
+    it('Makes an axios request when the user id is valid', async () => {
+      const validUserId = '5ce819935e539c343f141ece';
+
+      const response = await api.updateUserPassword(validUserId, 'password update');
+
+      expect(response.data).toEqual(mockResponse);
+    });
+  });
+
+  describe('getUser', () => {
+    const mockResponse = 'Mock user';
+    beforeAll(() => {
+      mockAxios.reset();
+      const url = /^.*\/v1\/users\/.*$/;
+      mockAxios.onPut(url).reply(200, mockResponse);
+    });
+
+    it('Returns an error when a url traversal is supplied', async () => {
+      const urlTraversal = '../../../etc/stealpassword';
+      const expectedResponse = { status: 400, data: 'Invalid user id' };
+
+      const response = await api.updateUserPassword(urlTraversal, 'password update');
+
+      expect(response).toMatchObject(expectedResponse);
+    });
+
+    it('Returns an error when a local IP is supplied', async () => {
+      const localIp = '127.0.0.1';
+      const expectedResponse = { status: 400, data: 'Invalid user id' };
+
+      const response = await api.updateUserPassword(localIp, 'password update');
+
+      expect(response).toMatchObject(expectedResponse);
+    });
+
+    it('Makes an axios request when the user id is valid', async () => {
+      const validUserId = '5ce819935e539c343f141ece';
+
+      const response = await api.updateUserPassword(validUserId, 'password update');
+
+      expect(response.data).toEqual(mockResponse);
+    });
+  });
+
+  describe('createFacilityAmendment', () => {
+    const mockResponse = 'Mock amendment';
+    beforeAll(() => {
+      mockAxios.reset();
+      const url = /^.*\/v1\/facilities\/.*\/amendments$/;
+      mockAxios.onPost(url).reply(200, mockResponse);
+    });
+
+    it('Returns an error when a url traversal is supplied', async () => {
+      const urlTraversal = '../../../etc/stealpassword';
+      const expectedResponse = { status: 400, data: 'Invalid facility id' };
+
+      const response = await api.createFacilityAmendment(urlTraversal, 'mock token');
```

Comment:
These hard coded credentials are test data only, happy for you to mark them all as `Uses in tests` @ChristianMcc